### ISSUE TITLE
Percentage based undotree_DiffpanelHeight

### ISSUE THREAD

**Issue by jamespharvey20** [mbbill/undotree#102]
I'd love to be able to specify `undotree_DiffpanelHeight` as a percentage, rather than a fixed number of lines.  I think I'd set it to 25% of the total height.  I run 90 degree rotated monitors and get 142 lines if they're maximized, or 74 lines if they're a bit shorter to reduce neck strain from looking upward.  I do switch these sizes often.  I also use tmux a lot, so the height of a vim window can be all over the place.  If I'm in a maximized terminal with a single vim, I'd love for the diff panel to take 35 of the 142 lines.  But, I can't set it that way because sometimes that can be more than the full height of that vim.  i'd also love for it to dynamically update upon resizing tmux pane size or total terminal size.

Only starting to get more familiar with vimscript, so I'm not sure this is possible (seems like it should be.)  It would be nice if to make `undotree_DiffpanelHeight` take either an integer or a percentage, and just detect the `%`.

**Comment by mbbill** [comment]
yeah, agreed. Setting width and height as percentage should be pretty straightforward. Dynamically adjusting panel width/height would be a bit trickier since it needs to hook up to some of the auto commands but I believe that's still doable.
